checks/runchecks.go

```diff
@@ -43,14 +33,14 @@ func RunChecks(ctx *context.Context) []*pkg.CheckResult {
 				switch v := r.Check.(type) {
 				case v1.TestFunction:
 					tpl := v.GetTestFunction()
-					if tpl.Template == "" {
+					if tpl.IsEmpty() {
 						break
 					}
 					message, err := template(ctx.New(r.Data), tpl)
 					if err != nil {
 						r.ErrorMessage(err)
 					} else if message != "true" {
-						r.Failf("")
+						r.Failf("expected test expression to be 'true' but got '%v'", message)
```

Comment:
I think for true or false results it shouldn't have any extra error message, if it is not true or false then you can have an error message, expecting either 'true' or 'false', got 'abc'